### DISCUSSION TITLE
Page to manage records

### DISCUSSION THREAD

**Discussion by gregoryloichot** [filamentphp/filament#13400] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2

### How can we help you?

Hi.

I'm triying to use sub navigation to edit relationships (https://filamentphp.com/docs/3.x/panels/resources/relation-managers#relation-pages).

The situation is : an article has many comments, a comments belongs to an article.

I created a "ManageCommentsPage" for the resource "Article". It is all good, when a click "Comments" in the sub-menu, all comments related to the articles are listed. 

The point is that the "Associate" action (to associate a comment to the current article) keep asking for an article**S** method in the Comment model (`Call to undefined method App\Models\Comment:articles()`). It does not exist because it is a belongsTo relationship, so article() not article**S**(). 

The Page is defined like :

```php
class ManageArticleComments extends ManageRelatedRecords
{
    protected static string $resource = ArticleResource::class;

    protected static string $relationship = 'comments';

    protected static ?string $navigationIcon = 'heroicon-o-rectangle-stack';

    public static function getNavigationLabel(): string
    {
        return 'Comments';
    }

    public function form(Form $form): Form
    {
        return CommentResource::form($form);
    }

    public function table(Table $table): Table
    {
        return CommentResource::table($table)
            ->recordTitleAttribute('name')
            ->filters([
                Tables\Filters\TrashedFilter::make()
            ])
            ->headerActions([
                Tables\Actions\CreateAction::make(),
                Tables\Actions\AssociateAction::make()->preloadRecordSelect(),
            ])
            ->actions([
                Tables\Actions\ViewAction::make(),
                Tables\Actions\EditAction::make(),
                Tables\Actions\DissociateAction::make(),
                Tables\Actions\DeleteAction::make(),
                Tables\Actions\ForceDeleteAction::make(),
                Tables\Actions\RestoreAction::make(),
            ])
            ->bulkActions([
                Tables\Actions\BulkActionGroup::make([
                    Tables\Actions\DissociateBulkAction::make(),
                    Tables\Actions\DeleteBulkAction::make(),
                    Tables\Actions\RestoreBulkAction::make(),
                    Tables\Actions\ForceDeleteBulkAction::make(),
                ]),
            ])
            ->modifyQueryUsing(fn(Builder $query) => $query->withoutGlobalScopes([
                SoftDeletingScope::class,
            ]));
    }
}
```

Any ideas?

**Comment by micraux** [comment]
Hello,

> It does not exist because it is a belongsTo relationship, so article() not articleS().

Why?

As far as I know, you can give your relation any name you want.

Check the name of your `BelongsToMany` relation in your `Comment` model.

**Reply by gregoryloichot** [reply, marked as answer]
Yes, that why I'm asking… 

I end up by adding `->inverseRelationship('article')` to the action…
